### PR TITLE
Adds Space Carp Sect

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

Sorry, I'm slightly obsessed with making Sects now... oops

Adds a Carp Sect to the Chaplain's sect choices
You can offer up meat for 20 favor each.
You can then spend that favor on either summoning a holy space-carp from the observer pool, summoning a space-carp suit, 
or spend it on "flooding" the chapel with water vapor!

Flood Area - Costs 200 favor - Takes 40 seconds
Summon Carp-Suit - Costs 300 favor - Takes 60 seconds
Summon Carp - Costs 500 - Takes 90 seconds
Meat - Gives 20 favor per meat

## Why It's Good For The Game

Yet another way to keep players in-round, but this time as holy space-carp!

As far as I can tell, without any admin responses to prayers, the Chaplain has almost nothing going on during most rounds.
Occasinally they get funerals, or get to fight a cult or revenant, and theres the flame and tech sects too, but this brings a bit more to the table, and hopefully crew will come by the chapel more often to interact with the chaplain and join the carp-cult!

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

Screenshot_3
Screenshot_4
Screenshot_2
Screenshot_10
Screenshot_11
Screenshot_5
Screenshot_6
Screenshot_7
Screenshot_8
Screenshot_9

</details>

## Changelog
:cl:
add: The Chaplain now has access to a Space Carp Sect!
add: Flood Area rite for the Carp Sect, flood the chapel with water - 200 Favor Cost
add: Summon Carp-Suit rite for the Carp Sect, Summons a Space-Carp suit! - 300 Favor cost
add: Summon Carp rite for the Carp Sect, Summon a Holy Space-Carp from the observer pool - 500 Favor cost
add: Carp Sect uses the blue altar
add: Summoned Carp are sub-chaplains and as such, are holy!
/:cl:

### CODE REVIEW

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm:289]: You increase it's maxHealth, but don't actually fill it up so it starts 100 hp away from it's max?

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	if(!length(candidates))
```

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	var/carp_species = pick(/mob/living/simple_animal/hostile/carp/megacarp, /mob/living/simple_animal/hostile/carp)
```
And delete the line above should work?

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	var/list/jobbans = list(ROLE_BRAINWASHED, ROLE_DEATHSQUAD, ROLE_DRONE, ROLE_LAVALAND, ROLE_MIND_TRANSFER, ROLE_POSIBRAIN, ROLE_SENTIENCE)
```
To be safe..

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	var/turf/altar_turf = get_turf(religious_tool)
```
To be safe..

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm:287]: Not sure if you actually have to set this manually.

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	carp.name = "Holy Space-Carp ([rand(1,999)])"
```
Save an unnecesarry var definition. 

Delete the line above.

File: code/modules/religion/religion_sects.dm
Comment (ivanmixo) [inline code/modules/religion/religion_sects.dm]: Early return here instead.

**Comment by TeomanTheGreat** [comment]
Extremely based. Praise Carp'Sie!

**Comment by EvilDragonfiend** [comment]
I'd say the carp suit should be replacing the suit appearance of an offered suit rather than summoning a new one.

File: code/modules/religion/religion_sects.dm
Comment (ro5490) [reply code/modules/religion/religion_sects.dm]: Slightly confused as to what you mean here

File: code/modules/religion/rites.dm
Comment (ro5490) [reply code/modules/religion/rites.dm:287]: Set what maunally? the name, ckey, or species?

File: code/modules/religion/religion_sects.dm
Comment (ro5490) [reply code/modules/religion/religion_sects.dm]: adjusting this completely, so it has the regular heal as well as the carp-blessing

Comment (ro5490) [reply code/modules/religion/religion_sects.dm]: Gave up on the above, it was refusing to work

**Comment by ro5490** [comment]
> Before merging that we should port sect improvements with uis from tg.

I'll look into this

File: code/modules/religion/rites.dm
Comment (ivanmixo) [reply code/modules/religion/rites.dm:287]: The ckey

File: code/modules/religion/religion_sects.dm
Comment (ivanmixo) [inline code/modules/religion/religion_sects.dm]:
```suggestion
	if(!isliving(L))
		return FALSE
	L.faction |= "carp"
	user.visible_message("<span class='notice'>[user] blessed [L] with the power of [GLOB.deity]! They are now protected from Space Carps, Although carps will still fight back if attacked.</span>")
	SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "blessing", /datum/mood_event/blessing)
	return TRUE
```

Also unsure if you really want this being a visible message to everyone.

File: code/modules/religion/rites.dm
Comment (ivanmixo) [inline code/modules/religion/rites.dm]:
```suggestion
	var/list/candidates = pollGhostCandidates("Do you wish to be summoned as a Holy Carp?", jobbans, null, FALSE)
```
Random comma.

File: code/modules/religion/rites.dm
Comment (ro5490) [reply code/modules/religion/rites.dm:287]: ckey was the only way I managed to get the player into the mob, for whatever reason transfer_mind and a few other bits didnt work

File: code/modules/religion/religion_sects.dm
Comment (ro5490) [reply code/modules/religion/religion_sects.dm]: Mainly kept it as a visible message so that people dont assume oh! antag! tbh

File: code/modules/religion/rites.dm
Comment (ivanmixo) [reply code/modules/religion/rites.dm:252]: Should be fine, nevermind.

**Comment by ro5490** [comment]
Currently putting this on hold until the decision on 
https://github.com/BeeStation/BeeStation-Hornet/pull/7194
Is concluded.
As I might need to re-do this a bit differently due to the changes

**Comment by ro5490** [comment]
Updating this to the TGUI version.
This PR now requires https://github.com/BeeStation/BeeStation-Hornet/pull/7194

or should do, at least

If not, I'll just re-PR these when that gets merged, got backup copies saved

**Comment by EvilDragonfiend** [comment]
actually, you had to make a branch from your master. you shouldn't merge and edit your master generally.

**Comment by EvilDragonfiend** [comment]
My suggestion: Save the change here to somewhere, delete your fork then fork Bee again, then make a branch from your new master, then make changes at the branch.

**Comment by EvilDragonfiend** [comment]
nevermind my words. I was sleepy and misread your PR. sorry. lol

**Comment by ro5490** [comment]
No worries, I just acicdentally merged the wrong things a few times, managed to fix it though

**Comment by ro5490** [comment]
This should be good to go!

**Comment by ro5490** [comment]
Reee
I cant remember how to remove a file from the PR, so if anyone can help let me know
> time to resolve yet another changelog conflict

**Comment by ro5490** [comment]
Hate the fact I rebased just to remove the changelog file but if it works it works

**Comment by ro5490** [comment]
or not. ffs

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Modifies changelog.html

**Comment by ro5490** [comment]
> Modifies changelog.html

Gonna have to re-make this PR, couldnt figure out where the changelog change came from rip

**Comment by ro5490** [comment]
Made a new PR because aaaaaaaaaaaaaaaaaaaaaaaaaaaaa
@PowerfulBacon 
https://github.com/BeeStation/BeeStation-Hornet/pull/7342

**Pull request closed without merging**
